src/Azure.Functions.Cli/Actions/HostActions/StartHostAction.cs

```diff
@@ -217,9 +225,14 @@ public override async Task RunAsync()
 
             var settings = SelfHostWebHostSettingsFactory.Create(Environment.CurrentDirectory);
 
-            (var listenUri, var baseUri, var certificate) = await Setup();
+            // Determine if middleware (Easy Auth) is enabled
+            var middlewareAuthSettings = _secretsManager.GetMiddlewareAuthSettings();
+            bool authenticationEnabled = middlewareAuthSettings.ContainsKeyCaseInsensitive(Constants.MiddlewareAuthEnabledSetting) &&
+                middlewareAuthSettings[Constants.MiddlewareAuthEnabledSetting].ToLower().Equals("true");
```

Comment:
This would be a bit cleaner:
```csharp
bool authenticationEnabled = 
    middlewareAuthSettings.TryGetValue(Constants.MiddlewareAuthEnabledSetting, out string enabledValue) &&
    bool.TryParse(enabledValue, out bool isEnabled) &&
    isEnabled;
```